### PR TITLE
Ingress interface standard

### PR DESCRIPTION
**Pull request by jdkandersson**
- Minor styling improvements
- Fix a broken unit test
- Add a unit test for the new interface standard and fix a bug it uncovered

### CODE REVIEW

File: lib/charms/nginx_ingress_integrator/v0/ingress.py [hunk 1]
```diff
@@ -117,7 +125,22 @@ class IngressRequires(Object):
...
-        self.framework.observe(charm.on["ingress"].relation_changed, self._on_relation_changed)
+        self.framework.observe(
+            charm.on["ingress"].relation_changed, self._on_relation_changed
+        )
+
+        # Set default values.
+        DEFAULT_RELATION_FIELDS = {
+            "service-namespace": self.model.name,
+        }
+        for k, v in DEFAULT_RELATION_FIELDS.items():
```
Comment (arturo-seijas) [inline lib/charms/nginx_ingress_integrator/v0/ingress.py:136]: Aren't the two statements in this if clause the same?

File: lib/charms/nginx_ingress_integrator/v0/ingress.py [hunk 2]
```diff
@@ -117,7 +125,22 @@ class IngressRequires(Object):
...
+            if k not in config_dict or not config_dict[k]:
+                config_dict[k] = v
+
+        # And now populate data for conformity with charm-relation-interfaces.
+        for k, v in RELATION_INTERFACES_MAPPINGS.items():
+            if k in config_dict and config_dict[k]:
```
Comment (arturo-seijas) [inline lib/charms/nginx_ingress_integrator/v0/ingress.py:142]: Do we need to check the existence of the key given this is validated afterwards in the `_config_dict_errors` method?

**Review by arturo-seijas — CHANGES_REQUESTED** [review]

**Review by mthaddon — COMMENTED** [review]
There seem to be quite a few formatting changes here. Can we remove those to make the diff easier to read, just including things that have changed (running `make blacken` doesn't suggest any formatting changes are needed).

As an example you're splitting this onto three lines even though it's less than the maximum line length:
`if field in BOOLEAN_CONFIG_FIELDS and (config_data is not None and config_data != ""):`

**Comment by jdkandersson** [comment]
> 



> There seem to be quite a few formatting changes here. Can we remove those to make the diff easier to read, just including things that have changed (running `make blacken` doesn't suggest any formatting changes are needed)?
> 
> As an example you're splitting this onto three lines even though it's less than the maximum line length: `if field in BOOLEAN_CONFIG_FIELDS and (config_data is not None and config_data != ""):`

I'll check, those were all done automatically by the code editor by running black

**Comment by jdkandersson** [comment]
> 



> > There seem to be quite a few formatting changes here. Can we remove those to make the diff easier to read, just including things that have changed (running `make blacken` doesn't suggest any formatting changes are needed)?
> > As an example you're splitting this onto three lines even though it's less than the maximum line length: `if field in BOOLEAN_CONFIG_FIELDS and (config_data is not None and config_data != ""):`
> 
> I'll check, those were all done automatically by the code editor by running black

So what happened is that the `make blacken` command used a line length of 99 as a parameter. The code editor used the default 80 and reformatted a bunch of code. I have now added the line length to `pyproject.toml` which means this shouldn't happen again.

File: lib/charms/nginx_ingress_integrator/v0/ingress.py (hunk 2 above)
Comment (jdkandersson) [reply lib/charms/nginx_ingress_integrator/v0/ingress.py:140]: I don't quite follow, this step is performing a mapping of the old keys to the new keys defined in the ingress standard to ensure compatibility with both. The code isn't very clear, I'll try and update it to make it clearer.

File: lib/charms/nginx_ingress_integrator/v0/ingress.py (hunk 1 above)
Comment (arturo-seijas) [reply lib/charms/nginx_ingress_integrator/v0/ingress.py]: Oh, yes. Makes sense. Nvm

**Review by arturo-seijas — APPROVED** [review]
LGTM

**Pull request merged**